src/libstd/sys/unix/thread.rs

```diff
@@ -125,16 +125,30 @@ impl Thread {
     }
 
     pub fn sleep(dur: Duration) {
-        let mut ts = libc::timespec {
-            tv_sec: dur.as_secs() as libc::time_t,
-            tv_nsec: dur.subsec_nanos() as libc::c_long,
-        };
+        fn get_secs(secs: &mut u64) -> libc::time_t {
+            let amt = cmp::min(libc::time_t::max_value() as u64, *secs);
+            *secs -= amt;
+            amt as libc::time_t
+        }
```

Comment:
This is small enough and only called in one location that this can probably be inlined directly below